src/betterproto/__init__.py

```diff
@@ -755,11 +876,112 @@ def _betterproto(self) -> ProtoClassMetadata:
             self.__class__._betterproto_meta = meta  # type: ignore
         return meta
 
+    def dump(self, stream: BinaryIO):
```

Comment:
It is worth syncing the fork and rebasing. I think the `oneof` pattern matching change modifies some parts of the method that this method is based on.